### PR TITLE
25545 scaffold auto coe page

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
This pull request adds a new route and page to the COE application. It performs the automated COE eligibility check and handles UI for displaying the certificate download to the user. 

## Testing done
- local, logged out, logged in, browser back button

## Screenshots
<details><summary>Mobile and Desktop views</summary>

Screen Shot 2021-06-11 at 2 13 07 PM
Screen Shot 2021-06-11 at 2 13 24 PM
</details

## Acceptance criteria
- [x] COE page renders when conditions are met, back button works.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/lgy/coe/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    setTimeout(() => {
+      resolve({ coe: true });
+    }, 2000);
+  });
+  //   return new Promise(reject => {
+  //     setTimeout(() => {
+  //       reject({ errors: [{ status: '500' }] });
+  //     }, 2000);
+  //   });
+};
```
Comment (micahchiang) [inline src/applications/lgy/coe/actions/index.js:18]: Replaced whenever  the endpoint in vets-api is ready.

File: src/applications/lgy/coe/containers/EligibilityApp.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  if (generateAutoCoeStatus === CALLSTATUS.idle || profileIsUpdating) {
+    content = <LoadingIndicator message="Loading application..." />;
+  } else if (generateAutoCoeStatus === CALLSTATUS.pending) {
+    content = (
+      <LoadingIndicator message="Checking automatic COE eligibility..." />
+    );
+  } else if (
+    generateAutoCoeStatus === CALLSTATUS.success ||
+    (generateAutoCoeStatus === CALLSTATUS.skip && coe)
```
Comment (micahchiang) [inline src/applications/lgy/coe/containers/EligibilityApp.jsx:44]: If a user has gone through the COE check, seen their COE but then opted to fill out the form, this allows them access back to their COE using the browser back button.

File: src/applications/lgy/coe/containers/IntroductionPage.jsx [hunk 3]
```diff
@@ -1,102 +1,41 @@
...
 import { focusElement } from 'platform/utilities/ui';
 import OMBInfo from '@department-of-veterans-affairs/component-library/OMBInfo';
-import FormTitle from 'platform/forms-system/src/js/components/FormTitle';
 import SaveInProgressIntro from 'platform/forms/save-in-progress/SaveInProgressIntro';
 
-class IntroductionPage extends React.Component {
-  componentDidMount() {
+import { FormApplicationSteps } from '../components/FormApplicationSteps';
+
+function IntroductionPage(props) {
```
Comment (micahchiang) [inline src/applications/lgy/coe/containers/IntroductionPage.jsx:9]: This file is largely just conversion from class to functional component with some refactoring.

**Review by Mottie — APPROVED** [review]
ghe-emoji

**Pull request merged**